src/stdlib/lib.c

```diff
@@ -54,15 +55,52 @@ struct subrange_t {
 
 typedef struct subrange_t* subrange_p;
 
-value_p get_val(range_p range, int row, int col);
+//value_p get_val(range_p range, int row, int col);
+
+value_p get_val(subrange_p range, int row, int col) {
+	//TODO: assertions
+	value_p val = range->range->values + (row + range->offsetRow) * range->range->cols + col + range->offsetCol;
+	return val;
+}
 
 int assertSingle(subrange_p range) {
 	return (range->subrangeRow == 1 && range->subrangeCol == 1);
 }
 
-subrange_p print(subrange_p whatever, subrange_p text) {
-	printf("%s", text->range->values->str->text);
-	subrange_p result = malloc(sizeof(struct subrange_t));
+int assertText(value_p my_val) {
+	return (my_val->flags == 2);
+}
+
+value_p empty() {
+	value_p empty_val = malloc(sizeof(struct value_t));
+	empty_val->flags = 0;
+	return empty_val;
+}
+
+value_p success() {
+	value_p val = malloc(sizeof(struct value_t));
+	val->flags = 1;
+	val->numericVal = 1;
```

Comment:
`numericVal` can be 0, since empty is separate and denotes errors. Simple matter of standardizing. The only reason I thought of 1 because of truthy numbers, but that's fine too!